blueocean-dashboard/src/main/java/io/jenkins/blueocean/BlueOceanWebURLBuilder.java

```diff
@@ -124,7 +124,7 @@ private BlueOceanWebURLBuilder() {
         } else if (classicModelObject instanceof Item) {
             Resource blueResource = BluePipelineFactory.resolve((Item) classicModelObject);
             if (blueResource instanceof BlueMultiBranchPipeline) {
-                return getOrgPrefix() + "/" + encodeURIComponent(((BluePipeline) blueResource).getFullName()) + "/branches";
+                return getOrgPrefix() + "/" + encodeURIComponent(decodeURIComponent(((BluePipeline) blueResource).getFullName())) + "/branches";
```

Comment:
yeah feels bad :( stapler has to do all sorts of things like this too (ie branches have to be double encoded for some reason, if you use classic you will see that). The proposed alternative was to totally change the branch name -> URI mapping (which stephen proposed) to something opaque. Certainly easier, but viscerally uglier.